hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/ClientJobProxy.java

```diff
@@ -156,16 +155,15 @@ protected long doGetJobSubmissionTime() {
     protected JobConfig doGetJobConfig() {
         return callAndRetryIfTargetNotFound(() -> {
             ClientMessage request = JetGetJobConfigCodec.encodeRequest(getId());
-            ClientMessage response = invocation(request, masterAddress()).invoke().get();
+            ClientMessage response = invocation(request, masterUuid()).invoke().get();
             Data data = JetGetJobConfigCodec.decodeResponse(response).response;
             return serializationService().toObject(data);
         });
     }
 
     @Override
-    protected Address masterAddress() {
-        Optional<Member> first = container().getCluster().getMembers().stream().findFirst();
-        return first.orElseThrow(() -> new IllegalStateException("No members found in cluster")).getAddress();
+    protected UUID masterUuid() {
+        return Util.getMaster(container().getCluster()).getUuid();
```

Comment:
you can use use clientClusterService.getMaster() here and inline the Util method

Reply:
Ok, but we should be aware that we are changing behaviour, not just adapting (ie. no more exception). Is that ok?